jobs/MeterInstanceJob.js

```diff
@@ -0,0 +1,186 @@
+'use strict';
+
+const _ = require('lodash');
+const Promise = require('bluebird');
+const logger = require('../common/logger');
+const BaseJob = require('./BaseJob');
+const catalog = require('../common/models/catalog');
+const config = require('../common/config');
+const utils = require('../common/utils');
+const maas = require('../data-access-layer/metering');
+
+const apiServerClient = require('../data-access-layer/eventmesh').apiServerClient;
+const CONST = require('../common/constants');
+
+class MeterInstanceJob extends BaseJob {
+  static run(job, done) {
+    return Promise.try(() => {
+      logger.debug('Starting MeterInstanceJob Job');
+      this.getInstanceEvents()
+        .tap(events => logger.info('recieved events -> ', events))
+        .then(events => this.meter(events))
+        .then((meterResponse) => this.runSucceeded(meterResponse, job, done))
+        .catch((err) => this.runFailed(err, {}, job, done));
+    });
+  }
+
+  static getInstanceEvents() {
+    const options = {
+      resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.INSTANCE,
+      resourceType: CONST.APISERVER.RESOURCE_TYPES.SFEVENT,
+      query: {
+        labelSelector: `meter_state in (${CONST.METER_STATE.TO_BE_METERED},${CONST.METER_STATE.FAILED})`
+      }
+    };
+    return apiServerClient.getResources(options);
+  }
+
+  static meter(events) {
+    logger.info(`Number of events to be metered in this run - ${events.length}`);
+    return Promise.try(() => {
+      let successCount = 0,
+        failureCount = 0,
+        failedEvents = [];
+      return Promise.map(events, (event) => {
+          return this
+            .sendEvent(event)
+            .then((status) => {
+              if (status) {
+                successCount++;
+              } else {
+                failureCount++;
+                failedEvents.push(event);
+              }
+            })
+            .catch(err => {
+              logger.error(`Error occurred while metering event : `, err);
+            });
+        }).then(() => {
+          return {
+            totalEvents: events.length,
+            success: successCount,
+            failed: failureCount,
+            failedEvents: failedEvents
+          };
+        })
+        .catch(err => {
+          logger.error(`Error occurred while metering all events : `, err);
+        });
+
+    });
+  }
+
+  static isServicePlanExcluded(options) {
```

Comment:
Such a complex logic will not be needed if we have the metadata part in the individual plan...right?

Reply:
How can we achieve landscape specific exclusion ?

Reply:
Since we decided to use plan level exclusion the code is modified as below